library/std/src/collections/hash/map.rs

```diff
@@ -243,8 +244,13 @@ use crate::ops::Index;
 #[cfg_attr(not(test), rustc_diagnostic_item = "HashMap")]
 #[stable(feature = "rust1", since = "1.0.0")]
 #[rustc_insignificant_dtor]
-pub struct HashMap<K, V, S = RandomState> {
-    base: base::HashMap<K, V, S>,
+pub struct HashMap<
+    K,
+    V,
+    S = RandomState,
+    #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global,
```

Comment:
Note that there is an `Allocator` bound on the type parameter. `S = RandomState` does not use a similar bound. However, hashbrown currently has this bound, so if we want to relax it to allow users to refer to non-well-formed `HashMap` types with nonsense allocator type parameters, that will first need to be changed upstream.